luthier/src/main/java/com/yahoo/bard/webservice/config/luthier/factories/KeyValueStoreDimensionFactory.java

```diff
@@ -96,10 +96,8 @@ public Dimension build(String name, ObjectNode configTable, LuthierIndustrialPar
         String longName = configTable.get("longName").textValue();
         String category = configTable.get("category").textValue();
         String description = configTable.get("description").textValue();
-        KeyValueStore keyValueStore = resourceFactories.getKeyValueStore(configTable.get("keyValueStore").textValue());
-        SearchProvider searchProvider = resourceFactories.getSearchProvider(
-                configTable.get("domain").textValue()
-        );
+        KeyValueStore keyValueStore = resourceFactories.getKeyValueStore(configTable.get("domain").textValue());
+        SearchProvider searchProvider = resourceFactories.getSearchProvider(configTable.get("domain").textValue());
```

Comment:
I vaguely remember why we're doing this, but it's going to be very confusing for people. So we need to make very sure we document this. We should document this in `dimensions.lua` and on the DimensionFactory
interface as part of the contract format that @michael-mclawhorn suggested.

Reply:
We don't have to this yet, but is definitely something we need to mention once we've figured what we want to the contract format to look like.

Reply:
Agreed. We can start that as a parallel task, and now is definitely a reasonable time for us to start.